src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3274 @@
+        {
+            Initialize(null, null);
+        }
+
+        /// <summary>
+        /// Initialize the class cache with the default classes in $ENV:SystemDirectory\Configuration.
+        /// </summary>
+        /// <param name="errors">Collection of any errors encountered during initialization.</param>
+        /// <param name="modulePathList">List of module path from where DSC PS modules will be loaded.</param>
+        public static void Initialize(Collection<Exception> errors, List<string> modulePathList)
+        {
+            s_tracer.WriteLine("Initializing DSC class cache force={0}");
+
+            if (Platform.IsLinux || Platform.IsMacOS)
+            {
+                //
+                // Load the base schema files.
+                //
+                ClearCache();
+                var dscConfigurationDirectory = Environment.GetEnvironmentVariable("DSC_HOME") ??
+                                                "/etc/opt/omi/conf/dsc/configuration";
+
+                if (!Directory.Exists(dscConfigurationDirectory))
+                {
+                    throw new DirectoryNotFoundException("Unable to find DSC schema store at " + dscConfigurationDirectory + ". Please ensure PS DSC for Linux is installed.");
+                }
+
+                var resourceBaseFile = Path.Combine(dscConfigurationDirectory, "BaseRegistration/BaseResource.schema.json");
+                ImportClasses(resourceBaseFile, s_defaultModuleInfoForResource, errors);
+                var metaConfigFile = Path.Combine(dscConfigurationDirectory, "BaseRegistration/MSFT_DSCMetaConfiguration.json");
+                ImportClasses(metaConfigFile, s_defaultModuleInfoForResource, errors);
+
+                var allResourceRoots = new string[] { dscConfigurationDirectory };
+
+                //
+                // Load all of the system resource schema files, searching
+                //
+                string resources;
+                foreach (var resourceRoot in allResourceRoots)
+                {
+                    resources = Path.Combine(resourceRoot, "schema");
+                    if (!Directory.Exists(resources))
+                    {
+                        continue;
+                    }
+
+                    foreach (var schemaFile in Directory.EnumerateDirectories(resources).SelectMany(d => Directory.EnumerateFiles(d, "*.schema.json")))
+                    {
+                        ImportClasses(schemaFile, s_defaultModuleInfoForResource, errors);
+                    }
+                }
+
+                // Linux DSC Modules are installed to the dscConfigurationDirectory, so no need to load them.
+            }
+            else
+            {
+                // DSC SxS scenario
+                var configSystemPath = Utils.DefaultPowerShellAppBase;
+                var systemResourceRoot = Path.Combine(configSystemPath, "Configuration");
+                var inboxModulePath = "Modules\\PSDesiredStateConfiguration";
```

Comment:
This is windows only?

Reply:
Yes; this is copy-paste of current mof-version of the code.